splash/src/main/kotlin/edu/artic/splash/SplashViewModel.kt

```diff
@@ -1,31 +1,48 @@
 package edu.artic.splash
 
 import android.util.Log
+import com.fuzz.rx.asObservable
+import com.fuzz.rx.bindTo
 import com.fuzz.rx.disposedBy
-import edu.artic.viewmodel.BaseViewModel
 import edu.artic.db.AppDataManager
-import edu.artic.db.AppDataState
+import edu.artic.db.ProgressDataState
+import edu.artic.viewmodel.NavViewViewModel
+import edu.artic.viewmodel.Navigate
+import io.reactivex.subjects.PublishSubject
 import javax.inject.Inject
 
-class SplashViewModel @Inject constructor(appDataManager : AppDataManager) : BaseViewModel() {
+class SplashViewModel @Inject constructor(appDataManager: AppDataManager) : NavViewViewModel<SplashViewModel.NavigationEndpoint>() {
+
+    sealed class NavigationEndpoint {
```

Comment:
I dont think this goes here? it should belong somewhere more global like a `nav-actions` module or something that other classes if need be can depend on that directly.

Reply:
so NavigationEndpoint is something very specific to a ViewModel in our current implementation of the navigation. so each view model will only have stuff like this for now. We can change it later, but mostly because we don't get how your router stuff works right now.